packages/cli/src/util/link/setup-and-link.ts

```diff
@@ -176,13 +188,12 @@ export default async function setupAndLink(
         deployStamp: stamp(),
         target: undefined,
         skipAutoDetectionConfirmation: false,
+        projectSettings: {
+          ...localConfigurationOverrides,
+          sourceFilesOutsideRootDirectory,
+        },
       };
 
-      if (isZeroConfig) {
-        // Only add projectSettings for zero config deployments
-        createArgs.projectSettings = { sourceFilesOutsideRootDirectory };
-      }
```

Comment:
I think we still need this.

Project Settings are not applied when `builds` prop is present.

Reply:
This whole thing is already wrapped in a if(isZeroConfig) though